lib/nerves_network/wifi_manager.ex

```diff
@@ -280,6 +280,19 @@ defmodule Nerves.Network.WiFiManager do
     %Nerves.Network.WiFiManager{state | wpa_pid: pid}
   end
 
+  defp parse_settings(settings) when is_list(settings) do
+    settings
+    |> Map.new
+    |> parse_settings
+  end
+
+  defp parse_settings(settings = %{ key_mgmt: key_mgmt }) when is_binary(key_mgmt) do
+    %{ settings | key_mgmt: String.to_atom(key_mgmt) }
```

Comment:
I don't think this is the _greatest_ solution.. This is going to lead to use needing to support both string and atom keys. The idea is good - But i think we might need to expand on the `parse_settings` function here, and instead of casting to a atom from a string, we should enforce types here by raising an ArgumentError if bad settings are sent.

Reply:
might i suggest adding another clause 
```elixir
defp parse_settings(%{ key_mgmt: key_mgmt }) when is_atom(key_mgmt) do
   raise ArgumentError ":key_mgmt setting must be a string"
end
```

The other problem is we may  be doing this api backwards. I know the goal was to allow `key_mgmt` to be inputted as a string, but i think having a list of aloud atoms would be just fine. (and preferable in my opinion). A list of aloud binaries would work for me too. I just don't think we should  be allowing any arbitrary argument to be passed here since there is a finite list of available `key_mgmt` settings.
ping @fhunleth

Reply:
_AAAANNND_ i just took a read back thru #17. 
@fhunleth said we _should_ initially support both `atom` and `string` keys. I would like long term to depreciate one or the other. Since There is a limited number of `key_mgmt` possibilities, i would personally prefer to keep the atom syntax, but others may disagree here. 

Approving for now, and opening a new issue for future updates.